src/main/java/net/guides/springboot2/springboot2jpacrudexample/service/EmployeeService.java

```diff
@@ -0,0 +1,67 @@
+package net.guides.springboot2.springboot2jpacrudexample.service;
+
+import net.guides.springboot2.springboot2jpacrudexample.exception.ResourceNotFoundException;
+import net.guides.springboot2.springboot2jpacrudexample.model.Employee;
+import net.guides.springboot2.springboot2jpacrudexample.repository.EmployeeRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+
+import javax.validation.Valid;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+@Service
+public class EmployeeService {
+    @Autowired
+    private EmployeeRepository employeeRepository;
+
+    public List<Employee> getAllEmployees() {
+        List<Employee> employees = employeeRepository.findAll();
+        return employees;
+    }
+
+    public ResponseEntity<Employee> getEmployeeById(@PathVariable(value = "id") Long employeeId)
+            throws ResourceNotFoundException {
+        Optional<Employee> employeeOptional = employeeRepository.findById(employeeId);
+        Employee employee = employeeOptional.orElseThrow(() -> new ResourceNotFoundException("Employee not found for this id :" + employeeId));
+        return ResponseEntity.ok().body(employee);
+    }
+
+    public Employee createEmployee(@Valid @RequestBody Employee employee) {
+        employee = employeeRepository.save(employee);
+        return employee;
+    }
+
+
+    public ResponseEntity<Employee> updateEmployee(@PathVariable(value = "id") Long employeeId,
```

Comment:
No need of @PathVariable and all. This is for getting values passed form UI in the controller class